PeerioTechnologies/peerio-desktop
# style: 'person' icon in Rooms header + update avatar in ProfileSettings

#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/888/desktop-screen-room-user-icon
https://app.clubhouse.io/peerio/story/1550/desktop-improve-error-handling-when-updating-avatar

#### Testing instructions  
The main visible change: In room, the "person" icon is no longer a button that opens the sidebar. Instead is just an icon, mouseover shows a tooltip that lists all members of the room.

PR includes a small update to `<Avatar>` component for `size="full"`, and using `<Avatar>` in ProfileSettings view. This allows the ProfileSettings avatar image to retry if download fails. (Before it was a `background-image`, and those don't retry.)

It is still possible to break avatar images by uploading a large image and then breaking internet connection a lot during the download. A broken image will be cached by Electron and will be persistent after restart. However, IMO this problem is out of scope for this particular story, and has more to do with image retrying & caching in general (maybe even at the Electron level?), not specific to avatars.

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
